utils/src/main/scala/za/co/absa/enceladus/utils/general/ConfigReader.scala

```diff
@@ -31,4 +35,64 @@ object ConfigReader {
   def readStringConfig(path: String, default: String): String = {
     readStringConfigIfExist(path).getOrElse(default)
   }
+
+  /**
+   * Flattens TypeSafe config tree and returns the effective configuration
+   * while redacting sensitive keys.
+   *
+   * @param sensitiveKeys A set of keys for which values shouldn't be logged.
+   * @return the effective configuration as a map
+   */
+  def getFlatConfig(sensitiveKeys: Set[String] = Set()): Map[String, String] = {
```

Comment:
I keep wondering whether to have the "flat config" is part of the requirement for being able to print the (redacted) config or not.

If not, maybe it could be easier to construct a `redactingConfig` from the `sensitiveKeys` and then just merge those two together yielding the resulting `redactedConfig` (still of the type `Config`!) that we could serialize using the usual means:
```scala
    // `regularConfig` is ultimately loaded by ConfigFactory.load()

    def addToConfigIfIn(ref: Config)(c: Config, key: String) = if (ref.hasPath(key)) {
      c.withValue(key, ConfigValueFactory.fromAnyRef("*****"))
    } else c
    val redactingConfig = sensitiveKeys.foldLeft(ConfigFactory.empty)(addToConfigIfIn(regularConfig))

    val redactedConfig = redactingConfig.withFallback(regularConfig)

   //... then the redactedConfig can be worked with. Probably serialized and printed using e.g.:
    redactedConfig.root().render(ConfigRenderOptions.concise)

```

On the first look, this seems easier than construct the flatConfig, but I might be missing something.

Reply:
I like this solution a lot! Looks very clean and with minimal dependency on `Config` implementation internals.

Reply:
Implemented almost exactly like this with the difference that it will render it in HOCON instead of JSON. 
Displaying the effective config in HOCON instead of JSON looks odd, but it might be that I'm too used to the old school properties syntax.